docs/site/migration/auth/example.md

```diff
@@ -0,0 +1,570 @@
+
+Based on the screenshot, suppose u2(user with id 2) makes a request to
+'/projects/1/donate', the authorization system will send subject as u2, object
+as project1, action as donate to casbin. And since u2 inherits role p1_team, and
+p1_team can perform donate on project1, casbin system returns allow to the
+authorization system.
+
+### Migrating Authorization
+
+First let's migrate the ACLs. The access control information are considered as
+authorization metadata, provided in decorator `@authorize()`. Go back the
+Project controller file, we can now specify the following fields for each
+endpoint:
+
+- resource: the resource name, it's 'project' in this controller
+- scopes: it maps to the action in the casbin policy
+- allowedRoles: since the casbin system handles the role resolving, this field
+  is for optimizing the comparison scope. E.g. when executing the enforcer to
+  make decision, only compares with the policies in the admin policy file.
+
+To make the tutorial concise, the code details is omitted here. You can find the
+ACLs and how the endpoints are decorated in file
+[src/controller/project.controller.ts](https://github.com/strongloop/loopback-next/blob/example/acl-migration/examples/access-control-migration/src/controllers/project.controller.ts)
+
+Then we write the authorizer that calls casbin enforcers to make decision
+
+(Since the PR is still in review, I will update the part when we agree on the
+details)
+
+- Create an authorizer that retrieves the authorization metadata from context,
+  then execute casbin enforcers to make decision
+- Create a voter for instance level endpoints to append the project id to the
+  resource name
+- Create casbin enforcers
+- Write casbin model and policies
+- Casbin persistency and synchronize(2nd Phase implementation, TBD)
+  - when create a new project
+    - create a set of p, project\${id}\_owner, action policies
+    - create a set of p, project\${id}\_team, action policies
+  - add a new user to a team
+    - find the projects owned by the team owner, then create role inherit rules
+      g, u${id}, project${id}\_team
+  - add a new endpoint(operation)
+    - for each of its allowed roles, add p, \${role}, action policy
+
+## Seed Data
+
+The application has pre-created data to try each role's permission. The original
+example seeds data in the boot script, now they are migrated to an observer file
+called `sample.observer.ts`.
+
+_Since the data are already generated in `db.json`, that observer file is
+skipped by default._
+
+## Try Out
+
+(a question for reviewers: do you prefer to see all the screenshots in this
+section? To be concise I tend to only describe by words, while if the flow is
+not clear enough, I can add pictures)
+
+Start the application:
+
+- Run `npm start` to start the application.
+- Open the explorer
+
+Try role 'admin':
+
+- Login as admin first (user Bob)
```

Comment:
The screenshot(s) i have in mind:
- under the `users/login` endpoint, it shows:
  - the request body json
  - the token as the response
- Authorize button
   - (optional) shows what it looks like after clicking the Authorize button

Suggestion: We could possibly highlight the portion of the screenshot with red rectangle, annotated with numbers that shows the sequence of actions that you're describing in text.